caffe2/core/operator.h

```diff
@@ -116,12 +116,20 @@ class CAFFE2_API OperatorBase : public Observable<OperatorBase> {
     return outputs_.at(idx)->template GetMutable<T>();
   }
 
+  Tensor* Output(
+      int idx,
+      const vector<int64_t>& dims,
+      const at::TensorOptions& options) {
+    return BlobGetMutableTensor(outputs_.at(idx), dims, options);
```

Comment:
Here, you'd want to do some sort of "if device is not set, set it to `context_`." You'd have to make a copy of TensorOptions, but it's only two words, not too expensive :)

Reply:
or maybe just make argument `at::TensorOptions options`? since we are defaulting to "device is not set" in the callsite.